zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java

```diff
@@ -2294,7 +2305,9 @@ public class WebSocketServiceCallback<T> extends SimpleServiceCallback<T> {
     @Override
     public void onFailure(Exception ex, ServiceContext context) throws IOException {
       super.onFailure(ex, context);
-      if (ex instanceof ForbiddenException) {
+      if (ex instanceof JobManagerForbiddenException) {
```

Comment:
In my opinion, this is the wrong place.
Please override the `onFailure` method in the `JobManagerServiceCallback` class.